code/datums/components/combo_attacks.dm

```diff
@@ -86,7 +83,9 @@
 /datum/component/combo_attacks/proc/on_attack(datum/source, mob/living/target, mob/user, click_parameters)
 	SIGNAL_HANDLER
 
-	if((target.stat == DEAD  && !work_on_corpses) || (target == user && !work_on_self) || HAS_TRAIT(user, TRAIT_PACIFISM))
+	if(!can_attack_callback?.Invoke(user, target))
```

Comment:
This will return `NONE` if `can_attack_callback` is null. If this isn't desired, change this check to `can_attack_callback && !can_attack_callback.Invoke(user, target)`